### ISSUE TITLE
go-guru.el: go-guru-set-scope depends on subr-x, but does not 'require' it

### ISSUE THREAD

**Issue by bcmills** [dominikh/go-mode.el#258]
Whenever I try to evaluate `M-x go-guru-set-scope`, I get the following error:
```
go-guru-set-scope: Symbol’s function definition is void: string-join
```

`M-x version` reports:
```
GNU Emacs 26.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.24), modified by Debian
```

From what I can tell, `string-join` comes from the `subr-x` package, and `go-guru-set-scope` succeeds if I explicitly evaluate `(require 'subr-x)` first.

Per the advice in `subr-x`¹, I believe that `go-guru.el` should include the line
```
(eval-when-compile (require 'subr-x))
```

¹https://github.com/emacs-mirror/emacs/blob/9f58ed156974f6345a85da419c57c4235f549f2a/lisp/emacs-lisp/subr-x.el#L33-L34

**Comment by bcmills** [comment]
(CC @jvshahid)

**Comment by jvshahid** [comment]
I wonder why I didn't run into this issue, probably the order in which I load features in my init file. @bcmills do you want me to submit a PR ?

**Comment by bcmills** [comment]
> @bcmills do you want me to submit a PR ?

Would be nice. 🙂

(I have a local workaround, but it seems like a good idea to fix this for other users, and since I install via MELPA I don't have a good setup for testing fixes.)

**Comment by jvshahid** [comment]
created #259 to address this issue

**Comment by konradjniemiec** [comment]
Wrote this in the pull request as well, but in order to improve backwards compatibility, we could just remove `string-join` and instead using `mapconcat 'identity`. `string-join` [is only available in Emacs >25.3](https://www.emacswiki.org/emacs/ElispCookbook#toc8)

**Comment by jvshahid** [comment]
I'll reply on the PR just to consolidate everything related to this specific PR in one place.
